src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/AppInsights/WebJobsTelemetryInitializer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    internal class WebJobsTelemetryInitializer : ITelemetryInitializer
+    {
+        internal const string AzureWebsiteName = "WEBSITE_SITE_NAME";
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            // RoleName is the app name
+            telemetry.Context.Cloud.RoleName = Environment.GetEnvironmentVariable(AzureWebsiteName) ?? "[Unknown]";
```

Comment:
It looks like it's not. I created a staging slot and my WEBSITE_SITE_NAME matched on both slots. I'll follow along with what you guys do here: https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/91016d62f3181e10d4cf589ef8fd64dadb6b54a2/Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs#L17